pinot-spi/src/main/java/org/apache/pinot/spi/utils/TimeUtils.java

```diff
@@ -194,4 +198,20 @@ public static boolean isPeriodValid(String timeStr) {
       return false;
     }
   }
+
+  public static final String DATE_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+
+  public static Long convertDateTimeStringToMillis(String timeStr) {
+    Long millis = 0L;
+    SimpleDateFormat f = new SimpleDateFormat(DATE_TIME_FORMAT);
+    if (timeStr != null) {
+      try {
+        Date d = f.parse(timeStr);
+        millis = d.getTime();
+      } catch (ParseException e) {
+        return ERROR_FLAG;
+      }
+    }
+    return millis;
```

Comment:
Please use `java.time` instead - you can replace this with 
```java
return timeStr == null ? 0L : Instant.parse(timeStr).toEpochMilli();
```

This handles ISO 8601 timestamps automatically.